Marlin/ultralcd_st7920_u8glib_rrd.h

```diff
@@ -43,20 +40,137 @@
 
 #include <U8glib.h>
 
+//set optimization so ARDUINO optimizes this file
+#pragma GCC optimize (3)
+
+#define DELAY_0_NOP  ;
+#define DELAY_1_NOP  __asm__("nop\n\t");
+#define DELAY_2_NOP  __asm__("nop\n\t" "nop\n\t");
+#define DELAY_3_NOP  __asm__("nop\n\t" "nop\n\t" "nop\n\t");
+#define DELAY_4_NOP  __asm__("nop\n\t" "nop\n\t" "nop\n\t" "nop\n\t");
+
+
+// If you want you can define your own set of delays in Configuration.h
+//#define ST7920_DELAY_1 DELAY_0_NOP
+//#define ST7920_DELAY_2 DELAY_0_NOP
+//#define ST7920_DELAY_3 DELAY_0_NOP
+
+#if F_CPU >= 20000000
+  #ifndef ST7920_DELAY_1
+    #define ST7920_DELAY_1 DELAY_0_NOP
+  #endif
+  #ifndef ST7920_DELAY_2
+    #define ST7920_DELAY_2 DELAY_0_NOP
+  #endif
+  #ifndef ST7920_DELAY_3
+    #define ST7920_DELAY_3 DELAY_1_NOP
+  #endif
+#elif (MOTHERBOARD == BOARD_3DRAG) || (MOTHERBOARD == BOARD_K8200)
+  #ifndef ST7920_DELAY_1
+    #define ST7920_DELAY_1 DELAY_0_NOP
+  #endif
+  #ifndef ST7920_DELAY_2
+    #define ST7920_DELAY_2 DELAY_3_NOP
+  #endif
+  #ifndef ST7920_DELAY_3
+    #define ST7920_DELAY_3 DELAY_0_NOP
+  #endif
+#elif (MOTHERBOARD == BOARD_MINIRAMBO)
+  #ifndef ST7920_DELAY_1
+    #define ST7920_DELAY_1 DELAY_0_NOP
+  #endif
+  #ifndef ST7920_DELAY_2
+    #define ST7920_DELAY_2 DELAY_4_NOP
+  #endif
+  #ifndef ST7920_DELAY_3
+    #define ST7920_DELAY_3 DELAY_0_NOP
+  #endif
+#elif (MOTHERBOARD == BOARD_RAMBO)
+  #ifndef ST7920_DELAY_1
+    #define ST7920_DELAY_1 DELAY_0_NOP
+  #endif
+  #ifndef ST7920_DELAY_2
+    #define ST7920_DELAY_2 DELAY_0_NOP
+  #endif
+  #ifndef ST7920_DELAY_3
+    #define ST7920_DELAY_3 DELAY_0_NOP
+  #endif
+#elif F_CPU == 16000000
+  #ifndef ST7920_DELAY_1
+    #define ST7920_DELAY_1 DELAY_0_NOP
+  #endif
+  #ifndef ST7920_DELAY_2
+    #define ST7920_DELAY_2 DELAY_0_NOP
+  #endif
+  #ifndef ST7920_DELAY_3
+    #define ST7920_DELAY_3 DELAY_1_NOP
+  #endif
+#else
+  #error "No valid condition for delays in 'ultralcd_st7920_u8glib_rrd.h'"
+#endif
```

Comment:
Could save a few lines here, defining the above like so instead:
```cpp
#define DEFAULT_ST7920_DELAY_1 DELAY_0_NOP
#define DEFAULT_ST7920_DELAY_2 DELAY_0_NOP
#define DEFAULT_ST7920_DELAY_3 DELAY_1_NOP
```
…and then after those are all defined…
```cpp
#ifndef ST7920_DELAY_1
  #define ST7920_DELAY_1 DEFAULT_ST7920_DELAY_1
#endif
#ifndef ST7920_DELAY_2
  #define ST7920_DELAY_2 DEFAULT_ST7920_DELAY_2
#endif
#ifndef ST7920_DELAY_3
  #define ST7920_DELAY_3 DEFAULT_ST7920_DELAY_3
#endif
```